pkg/noun/urth.c

```diff
@@ -728,9 +733,11 @@ u3u_mmap_read(c3_c* cap_c, c3_c* pat_c, c3_d* out_d, c3_y** out_y)
     void* ptr_v;
 
     if ( MAP_FAILED == (ptr_v = mmap(0, len_d, PROT_READ, MAP_SHARED, fid_i, 0)) ) {
+      int err_i = errno;
       fprintf(stderr, "%s: mmap failed (%s): %s\r\n",
-                      cap_c, pat_c, strerror(errno));
+                      cap_c, pat_c, strerror(err_i));
       close(fid_i);
+      errno = err_i;
       return c3n;
```

Comment:
I understand why you're assigning errno to a temporary variable here to prevent its value being clobbered by the fprintf, but we don't care if errno gets clobbered because we've effectively translated its value to c3n (the only vere status code we have for now) and the caller shouldn't care what its value was.

-- right? or am I reading this wrong

Reply:
Yeah, good catch.